src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorNetworkIT.java

```diff
@@ -68,6 +78,123 @@ public void cleanup() {
     stopConnect();
   }
 
+  /**
+   * Transformer that blocks all incoming requests until {@link #release(int)} is called
+   * to fairly unblock a given number of requests.
+   */
+  public static class ConcurrencyTransformer extends ResponseTransformer {
+
+    private final Semaphore s = new Semaphore(0, true);
+    private final AtomicInteger requestCount = new AtomicInteger();
+
+    public ConcurrencyTransformer() {
+      s.drainPermits();
```

Comment:
could you please educate me on if we have started with 0 permits then why is drainPermits required here?

Reply:
We don't, leftover from a previous version where we started with a few permits. Removed.